tutorials/Creating a new Lumigator endpoint.md

```diff
@@ -0,0 +1,581 @@
+# Creating a new Lumigator endpoint
+
+This document shows how current endpoints in lumigator work and describes the process needed to create a new one. This
+might or might not become part of the public documentation - I think it might be useful for users to expand the tool
+with their own use cases and would allow us to grow more easily, but I am happy to collect feedback about it here first.
+
+## Lumigator's architecture
+
+The diagram below shows the current architecture of Lumigator. The larger
+containers (UI, Backend, Ray cluster, DB, Storage) are the different services Lumigator relies on.
+The components inside Backend, instead, are the different abstraction layers the backend itself relies on:
+
+* The **API** makes backend functionalities available to the UI (or whatever front-end you want to develop) through
+  different **routes** (code
+  [here](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/api/routes)).
+  [**Schemas**](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/schemas) are used in the API
+  which allows one to exactly know which kind of data has to be passed to it (similarly, using the same schemas will be
+  a requirement for our SDK).
+
+* **Services** implement the actual functionalities and are called by the different methods exposed in the API (code
+  [here](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/services)).
+
+* **Repositories** implement the [repository pattern](https://www.cosmicpython.com/book/chapter_02_repository.html) as
+  an abstraction over the DB (code
+  [here](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/repositories)). They make use
+  of [record classes](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/records) to refer
+  to actual records on the DB.
+
+<p style="text-align: center;">
+<img src="assets/Lumigator%20Architecture%20-%2020240902.jpg" alt="Lumigator's architecture" >
```

Comment:
suggestion, let's use the one already in the repo, and if we don't like it let's update: https://github.com/mozilla-ai/lumigator/blob/main/notebooks/assets/platform.png

Reply:
Done! I provided that for the general architecture, and then left a detail of Backend just to show how the different classes interact